app/utils/base_client.js

```diff
@@ -79,18 +75,27 @@ const getHeaders = function getHeaders (args, segmentData) {
 const _request = function request (methodName, url, args, callback, subSegment) {
   const namespace = getNamespace(clsXrayConfig.nameSpaceName)
   const clsSegment = namespace ? namespace.get(clsXrayConfig.segmentKeyName) : null
-  const parsedUrl = urlParse.parse(url)
+  const parsedUrl = urlParse(url)
+
+  var urlOrForwardProxy
```

Comment:
rather than doing it like this I would prefer to encapsulate this logic in a function, so 
```
const addProxy = function addProxy (url) {
  const parsedUrl = urlParse(url)

  if (process.env.FORWARD_PROXY_URL) {
    const parsedProxyUrl = urlParse(process.env.FORWARD_PROXY_URL)
    return _.merge(parsedUrl, _.pick(parsedProxyUrl, ['hostname', 'port']))
  }
  return parsedUrl
}
```

then

```
const proxiedUrl = addProxy(url)
  const httpsOptions = {
    hostname: proxiedUrl.hostname,
    port: proxiedUrl.port,
    path: proxiedUrl.pathname,
...
```